pkg/controllers/customcluster_controller.go

```diff
@@ -241,55 +268,75 @@ func (r *CustomClusterController) reconcile(ctx context.Context, customCluster *
 	if !cluster.DeletionTimestamp.IsZero() {
 		// If customCluster is already in phase Deleting, the controller will check the terminating worker to handle Deleting process.
 		if phase == v1alpha1.DeletingPhase {
-			return r.reconcileHandleDeleting(ctx, customCluster, customMachine)
+			return r.reconcileHandleDeleting(ctx, customCluster, customMachine, kcp)
 		}
 		// If customCluster is not in Deleting, the controller should terminate the Vms cluster by create a terminating worker.
 		return r.reconcileVMsTerminate(ctx, customCluster)
 	}
 
 	// CustomCluster in phase nil or ProvisionFailed will try to enter Provisioning phase by creating an init worker successfully.
 	if phase == v1alpha1.PendingPhase || phase == v1alpha1.ProvisionFailedPhase {
-		return r.reconcileCustomClusterInit(ctx, customCluster, customMachine, cluster)
+		return r.reconcileCustomClusterInit(ctx, customCluster, customMachine, cluster, kcp)
 	}
 
 	// If customCluster is in phase Provisioning, the controller will handle Provisioning process by checking the init worker status.
 	if phase == v1alpha1.ProvisioningPhase {
-		return r.reconcileHandleProvisioning(ctx, customCluster, customMachine, cluster)
+		return r.reconcileHandleProvisioning(ctx, customCluster, customMachine, cluster, kcp)
 	}
 
-	// Handle worker nodes scaling
-	if phase == v1alpha1.ProvisionedPhase || phase == v1alpha1.ScalingUpPhase || phase == v1alpha1.ScalingDownPhase {
-		currentClusterInfo := getClusterInfoFromCustomMachine(customMachine)
-		provisionedClusterInfo, err := r.getProvisionedClusterInfoFromConfigmap(ctx, customCluster)
-		if err != nil {
-			log.Error(err, "failed to get provisioned cluster Info from configmap")
-			return ctrl.Result{RequeueAfter: RequeueAfter}, err
+	// Obtain desiredClusterInfo and provisionedClusterInfo to determine if further scaling or upgrading is necessary.
+	desiredClusterInfo := getDesiredClusterInfo(customMachine, kcp)
+	provisionedClusterInfo, err1 := r.getProvisionedClusterInfo(ctx, customCluster)
+	if err1 != nil {
+		log.Error(err1, "failed to get provisioned cluster Info from configmap")
+		return ctrl.Result{RequeueAfter: RequeueAfter}, err1
```

Comment:
RequeueAfter does not take effect when err happens

Reply:
ok